src/definitions.jl

```diff
@@ -680,15 +680,31 @@ adjoint_mul(p::Plan, x::AbstractArray) = adjoint_mul(p, x, AdjointStyle(p))
 function adjoint_mul(p::Plan{T}, x::AbstractArray, ::FFTAdjointStyle) where {T}
     dims = fftdims(p)
     N = normalization(T, size(p), dims)
-    return (p \ x) / N
+    pinv = inv(p)
+    # Optimization: when pinv is a ScaledPlan, check if we can avoid a loop over x.  
+    # Even if not, ensure that we do only one pass by combining the normalization with the plan.
+    if pinv isa ScaledPlan && isapprox(pinv.scale, N)
+        scaled_pinv = pinv.p
+    else
+        scaled_pinv = (1/N) * pinv
+    end
+    return scaled_pinv * x
 end
 
 function adjoint_mul(p::Plan{T}, x::AbstractArray, ::RFFTAdjointStyle) where {T<:Real}
     dims = fftdims(p)
     N = normalization(T, size(p), dims)
     halfdim = first(dims)
     d = size(p, halfdim)
-    n = size(inv(p), halfdim)
+    pinv = inv(p)
+    n = size(pinv, halfdim)
+    # Optimization: when pinv is a ScaledPlan, fuse the scaling into our map to ensure we do not loop over x twice.
+    if pinv isa ScaledPlan
+        N /= pinv.scale
```

Comment:
Maybe instead define `scale = 1 / normalization(...)` and then do `scale *= pinv.scale`, so that you can do a multiplication inside the loop rather than a division?